CS3219-AY2223S1/cs3219-project-ay2223s1-g58
# Update user-service code & CI

**What is the purpose of this pull request?**


**Overview of changes**
Closes #25, Closes #26
- Fix a bug in frontend for retrieving access token
- Update CI to run lint & test in docker
- Add delete and update endpoints
- Add logging
- Change access token to expire in 60 seconds to test the frontend ability to refresh
- Add a basic test
- Some refactoring
- Adjust the development guide (see README in user-service)
  - due to it being slightly troublesome to map docker volume on a Windows machine (only works well if the folder is in wsl), I am changing the development setup for now to be:
    - run express server locally
    - run mongo and redis in docker
    - see instructions in readme

Testing:
- See that the CI run is successful
- Check that the development guide works, run the whole setup locally, then run `npm run start` in frontend to see if you can still play around with the register, login, logout

**Service(s) Touched**
